source/world_builder/utilities.cc

```diff
@@ -1405,22 +1440,86 @@ namespace WorldBuilder
             compare_point2[1] = coordinate_system->natural_coordinate_system() == cartesian ? Pb2[1] : Pb2[0];
             compare_point2[2] = coordinate_system->natural_coordinate_system() == cartesian ? position_in_natural_coordinates_at_min_depth.get_depth_coordinate() : Pb2[1];
 
-            distance_ridge = std::min(distance_ridge,
-                                      coordinate_system->distance_between_points_at_same_depth(Point<3>(position_in_natural_coordinates_at_min_depth.get_coordinates(),
-                                          position_in_natural_coordinates_at_min_depth.get_coordinate_system()),
-                                          compare_point1));
+            const double compare_distance1 = coordinate_system->distance_between_points_at_same_depth(Point<3>(position_in_natural_coordinates_at_min_depth.get_coordinates(),
+                                             position_in_natural_coordinates_at_min_depth.get_coordinate_system()),
+                                             compare_point1);
 
-            distance_ridge = std::min(distance_ridge,
-                                      coordinate_system->distance_between_points_at_same_depth(Point<3>(position_in_natural_coordinates_at_min_depth.get_coordinates(),
-                                          position_in_natural_coordinates_at_min_depth.get_coordinate_system()),
-                                          compare_point2));
+            const double compare_distance2 = coordinate_system->distance_between_points_at_same_depth(Point<3>(position_in_natural_coordinates_at_min_depth.get_coordinates(),
+                                             position_in_natural_coordinates_at_min_depth.get_coordinate_system()),
+                                             compare_point2);
+
+            double compare_distance = compare_distance1;
+            double spreading_velocity_at_ridge_pt = spreading_velocity_at_ridge_pt1;
+            double subducting_velocity_at_trench_pt = subducting_velocity_at_trench_pt1;
+
+            if (compare_distance2 < compare_distance1)
```

Comment:
Could you add a comment somewhere what distance1 and distance2 represent, or make the name more clear? Secondly, it doesn't seem to be used in any of the tests, so could you add a comment on in what cases this is used, and/or add a testcase where it is used?